CMakeLists.txt

```diff
@@ -572,26 +572,30 @@ if(WIN32)
   add_compile_options(/Zl) # omit default library name in .OBJ
 endif(WIN32)
 
+#--------------------------------
+# Definition directives
+#  - all clr specific compile definitions should be included in this file
+#  - all clr specific feature variable should also be added in this file
+#----------------------------------
+include(clrdefinitions.cmake)
```

Comment:
Looking at this again I think I misunderstood the CMake file - I am building the GC PAL after the CoreCLR PAL, but before we are adding the CoreCLR PAL header directories to the inclusion paths. Adding the PAL headers to the inclusion path is what causes problems for the GC PAL, since the GC PAL is trying to use system headers.